server/sonar-web/src/main/js/api/metrics.ts

```diff
@@ -21,8 +21,34 @@ import { getJSON } from '../helpers/request';
 import { Metric } from '../app/types';
 import throwGlobalError from '../app/utils/throwGlobalError';
 
-export function getMetrics(): Promise<Metric[]> {
-  return getJSON('/api/metrics/search', { ps: 9999 }).then(r => r.metrics);
+interface MetricsResponse {
+  metrics: Metric[];
+  p: number;
+  ps: number;
+  total: number;
+}
+
+export function getMetrics(data: { p?: number; ps?: number }): Promise<MetricsResponse> {
+  return getJSON('/api/metrics/search', data).catch(throwGlobalError);
+}
+
+export function getAllMetrics(
+  data: { p?: number; ps?: number } = { ps: 500 },
+  prev?: MetricsResponse
```

Comment:
I don't think it's a good idea to expose this parameter. Instead create an inner function:
```javascript
export function getAllMetrics(data) {
  return inner(data);

  function inner(data, prev) { ... }
}
```